// ng2-material-dropdown/esm2020/src/modules/ng2-dropdown.module.mjs
import { Ng2Dropdown } from './components/dropdown/ng2-dropdown';
import { Ng2DropdownMenu } from './components/menu/ng2-dropdown-menu';
import { Ng2DropdownButton } from './components/button/ng2-dropdown-button';
import { Ng2MenuItem } from './components/menu-item/ng2-menu-item';
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { DropdownStateService } from './services/dropdown-state.service';
import * as i0 from "@angular/core";
export class Ng2DropdownModule {
}
Ng2DropdownModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: Ng2DropdownModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
Ng2DropdownModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: Ng2DropdownModule, declarations: [Ng2Dropdown,
        Ng2MenuItem,
        Ng2DropdownButton,
        Ng2DropdownMenu], imports: [CommonModule], exports: [Ng2MenuItem,
        Ng2DropdownButton,
        Ng2DropdownMenu,
        Ng2Dropdown] });
Ng2DropdownModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: Ng2DropdownModule, imports: [[
            CommonModule
        ]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.0.3", ngImport: i0, type: Ng2DropdownModule, decorators: [{
            type: NgModule,
            args: [{
                    exports: [
                        Ng2MenuItem,
                        Ng2DropdownButton,
                        Ng2DropdownMenu,
                        Ng2Dropdown
                    ],
                    declarations: [
                        Ng2Dropdown,
                        Ng2MenuItem,
                        Ng2DropdownButton,
                        Ng2DropdownMenu,
                    ],
                    imports: [
                        CommonModule
                    ]
                }]
        }] });
export { Ng2Dropdown, Ng2DropdownMenu, Ng2MenuItem, Ng2DropdownButton, DropdownStateService };
